### PR TITLE
add Gentoo support

### PR DESCRIPTION
**Pull request by derdanne**
add Gentoo support

### CODE REVIEW

File: manifests/package.pp [hunk 1]
```diff
@@ -24,154 +24,163 @@
...
-      creates => $elasticsearch::package_dir,
-    }
-
-    file { $elasticsearch::package_dir:
-      ensure  => 'directory',
-      purge   => $elasticsearch::purge_package_dir,
-      force   => $elasticsearch::purge_package_dir,
-      backup  => false,
-      require => Exec['create_package_dir_elasticsearch'],
+  case $::osfamily {
```
Comment (electrical) [inline manifests/package.pp]: Any specific reason for this? Why wouldn't the current code be able to handle it?

**Comment by derdanne** [comment]
Gentoo has elasticsearch as native package provided by portage - i would prefer to manage the installation the native way.

**Comment by electrical** [comment]
@derdanne by default it uses the native way for any distro, all the extra code is just optional when you want to use the package_url option, so you don't have to make it a case statement.

**Comment by derdanne** [comment]
Oh, i see - reverted the changes of package.pp

File: manifests/service/openrc.pp [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      if(has_key($init_defaults, 'ES_USER')) {
+        if($init_defaults['ES_USER'] != $elasticsearch::elasticsearch_user) {
+          fail('Found ES_USER setting for init_defaults but is not same as elasticsearch_user setting. Please use elasticsearch_user setting.')
+        }
+      }
+
+      $init_defaults_pre_hash = { 'ES_USER' => $elasticsearch::elasticsearch_user, 'ES_GROUP' => $elasticsearch::elasticsearch_group, 'MAX_OPEN_FILES' => '65535' }
+      $new_init_defaults = merge($init_defaults_pre_hash, $init_defaults)
+
+      augeas { "defaults_${name}":
```
Comment (electrical) [inline manifests/service/openrc.pp]: I think this block can be removed since the file isn't used anywhere? Or am i missing it?

Comment (derdanne) [inline manifests/service/openrc.pp]: The defaults file is used:

$defaults_location  = '/etc/conf.d' in params.pp 
and "${elasticsearch::params::defaults_location}/elasticsearch.${name}" matches with init file /etc/init.d/elasticsearch.${name}

Comment (electrical) [inline manifests/service/openrc.pp]: Ahh okay. i didn't see it in the openrc script. I guess its being read automatically then?

Comment (derdanne) [inline manifests/service/openrc.pp]: Yes, the defaults file is automatically parsed, when it has the same name as the init script.

Comment (electrical) [inline manifests/service/openrc.pp]: Ahh okay. much smarter then the init scripts :-)

**Comment by electrical** [comment]
Thanks for all the work so far. I'll let Jenkins do its magic and merge it when its green.

**Comment by electrical** [comment]
jenkins, test it

**Comment by electrical** [comment]
jenkins, test it

**Comment by electrical** [comment]
@derdanne would you be able to squash your commits into a single commit? Then i will merge it.
Thanks!

**Comment by derdanne** [comment]
Can you give me a hint how i can revert the merges from master? 

I just pushed the github button "merge changes from master", and i don't get this working :/

**Comment by electrical** [comment]
Ahh okay. you can do a `git rebase -i`  which gives you the option to pick and squash commits.
If you set the first one to pick and the other 2 to squash you can modify it all into a single commit.

**Comment by derdanne** [comment]
OK, that does not seem to work - then i get all commits but not the merges - i think i have to do a new pull request with my changes only :(

**Comment by electrical** [comment]
No worries :) if you could open a new pr then would be great.

Thanks.

**Comment by derdanne** [comment]
Ok, i've created a new PR https://github.com/elastic/puppet-elasticsearch/pull/559

**Pull request closed without merging**